tests/unittest/_torch/sampler/test_beam_search.py

```diff
@@ -1,43 +1,278 @@
+            checkpoint_dir: Directory containing configuration files
+            **kwargs: Additional loading parameters
+        Returns:
+            ModelConfig object containing parsed configuration
+        """
+        return ModelConfig(pretrained_config=DummyConfig())
+
+
 @pytest.fixture(scope="module")
 def input_prompts():
-    return [
-        "Born in north-east France, Soyer trained as a",
-        "The future of AI is",
-    ]
+    return [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
 
 
-@pytest.fixture(scope="module")
-def expected_outputs():
-    return {
-        "Born in north-east France, Soyer trained as a": [
-            "painter in Paris before moving to London in",
-            "painter and sculptor in Paris before moving"
-        ],
-        "The future of AI is":
-        ["bright, but it's not without", "bright, but it's not going"],
-    }
+class BeamSearchTestOutput:
+
+    def __init__(self, outputs: torch.Tensor, cache_indirection: torch.Tensor):
+        self.outputs = outputs
+        self.cache_indirection = cache_indirection
+
+
+def get_expected_outputs(
+        start_token: int,
+        num_iterations: int = 4,
+        vocab_size: int = 1000,
+        use_overlap_scheduler: bool = False) -> list[list[int]]:
+    """Get the expected outputs for the given start token, number of iterations, vocabulary size
+       This function only works for a beam width of 2.
+
+       arguments:
+       - start_token: the token to start the generation from. This is the last token of the input prompt.
+       - num_iterations: the number of iterations to generate
+       - vocab_size: the size of the vocabulary
+       - use_overlap_scheduler: whether to use the overlap scheduler
+       returns:
+       - BeamSearchTestOutput: a named tuple containing the expected outputs and cache indirection
+           - expected_outputs: the expected outputs for the given start token, number of iterations, vocabulary size
+           - expected_cache_indirection: the expected cache indirection for the given start token, number of iterations, vocabulary size
+    """
+    expected_outputs = []
+    expected_cache_indirection = []
+
+    # These scores are deterministically set to the 4 tokens following the current input token (see model forward pass)
+    base_scores = torch.tensor([0.1, 0.2, 0.3, 0.4])
+    # extend the base scores to the full vocabulary size to calculate the correct softmax scores
+    full_scores = torch.zeros(vocab_size)
+    full_scores[:base_scores.shape[0]] = base_scores
+
+    # calculate the softmax scores for the first token
+    softmax_score = torch.log_softmax(full_scores * start_token, dim=-1)
+    # get the top 2 offsets => these will become beam 0 and beam 1
+    # The base scores are at the start of the full scores. So we can use their index as offset.
+    top_offset, second_best_offset = torch.topk(softmax_score, k=2,
+                                                dim=-1).indices
+
+    # iteration 1
```

Comment:
I suggest changing that since the loop below starts at 1.
```suggestion
    # First iteration
```